ffi/diplomat/src/datetime.rs

```diff
@@ -5,244 +5,230 @@
 #[diplomat::bridge]
 pub mod ffi {
     use alloc::boxed::Box;
+    use alloc::sync::Arc;
+    use core::convert::TryInto;
+
     use diplomat_runtime::DiplomatResult;
-    use icu_calendar::Gregorian;
-    use icu_datetime::{
-        options::length, DateTimeFormatter, TimeFormatter, TypedDateFormatter,
-        TypedDateTimeFormatter,
-    };
+    use icu_calendar::types::Time;
+    use icu_calendar::AnyCalendar;
+    use icu_calendar::{DateTime, Gregorian, Iso};
 
-    use crate::{
-        calendar::ffi::ICU4XDateTime, calendar::ffi::ICU4XGregorianDateTime,
-        errors::ffi::ICU4XError, locale::ffi::ICU4XLocale, provider::ffi::ICU4XDataProvider,
-    };
+    use crate::calendar::ffi::ICU4XCalendar;
+    use crate::date::ffi::{ICU4XDate, ICU4XIsoDate};
+    use crate::errors::ffi::ICU4XError;
+    use crate::time::ffi::ICU4XTime;
 
     #[diplomat::opaque]
-    /// An ICU4X TimeFormatter object capable of formatting a [`ICU4XGregorianDateTime`] as a string
-    #[diplomat::rust_link(icu::datetime::TimeFormatter, Struct)]
-    // TODO(#2153) - Rename to ICU4XTimeFormatter when we remove the dependency on calendar
-    // from TimeFormatter.
-    pub struct ICU4XTimeFormatter(pub TimeFormatter);
-
-    pub enum ICU4XTimeLength {
-        Full,
-        Long,
-        Medium,
-        Short,
-    }
+    /// An ICU4X DateTime object capable of containing a Gregorian date and time.
+    #[diplomat::rust_link(icu::calendar::DateTime, Struct)]
+    pub struct ICU4XGregorianDateTime(pub DateTime<Gregorian>);
 
-    impl ICU4XTimeFormatter {
-        /// Creates a new [`ICU4XTimeFormatter`] from locale data.
-        #[diplomat::rust_link(icu::decimal::TypedDateFormatter::try_new_unstable, FnInStruct)]
+    impl ICU4XGregorianDateTime {
```

Comment:
As discussed offline: remove `ICU4XGregorianDateTime` and use `ICU4XIsoDateTime` in its place